### PR TITLE
fix: improve descriptions for config options

### LINKED ISSUE #15181: Fix `description` fields in `lib/config/options/index.ts`
### Describe the proposed change(s).

Source file to edit: [`lib/config/options/index.ts`](https://github.com/renovatebot/renovate/blob/main/lib/config/options/index.ts).

Todos:

- [x] Use `monospaced` font for terms like `true`, `false`. Use backticks.
- [x] End sentences with full stops `.`
- [x] Capitalize proper nouns
- [x] Use plain language

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Improve `descriptions` for config options

## Context

Closes #15181.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -42,7 +43,7 @@ const options: RenovateOptions[] = [
   {
     name: 'postUpgradeTasks',
     description:
-      'Post-upgrade tasks that are executed before a commit is made by Renovate.',
+      'Post-upgrade tasks that are executed before a commit is made by Renovate.', // What's the difference between `postUpgradeTasks` and the `commands` below?
```
Comment (rarkins) [inline lib/config/options/index.ts:46]:
```suggestion
      'Post-upgrade tasks that are executed before a commit is made by Renovate.',
```

commands are a subset of postUpgradeTasks, they exist "inside" it

File: lib/config/options/index.ts [hunk 2]
```diff
@@ -53,7 +54,7 @@ const options: RenovateOptions[] = [
   {
     name: 'commands',
     description:
-      'A list of post-upgrade commands that are executed before a commit is made by Renovate.',
+      'A list of post-upgrade commands that are executed before a commit is made by Renovate.', // See comment above.
```
Comment (rarkins) [inline lib/config/options/index.ts:57]:
```suggestion
      'A list of post-upgrade commands that are executed before a commit is made by Renovate.',
```

File: lib/config/options/index.ts [hunk 3]
```diff
@@ -101,31 +102,31 @@ const options: RenovateOptions[] = [
...
+      'Change this value to override the default onboarding PR title.',
     type: 'string',
     default: 'Configure Renovate',
     globalOnly: true,
     cli: false,
   },
   {
     name: 'productLinks',
-    description: 'Links which are embedded within PRs, issues, etc.',
+    description: 'Links which are used within PRs, issues.', // I don't think etc. applies here, are there other places where Renovate can put links???
```
Comment (rarkins) [inline lib/config/options/index.ts:129]:
```suggestion
    description: 'Links which are used within PRs, issues.',
```

Potentially in issue and PR comments, although I'm not sure if we do that

File: lib/config/options/index.ts [hunk 4]
```diff
@@ -163,7 +164,7 @@ const options: RenovateOptions[] = [
   {
     name: 'ignorePresets',
     description:
-      'A list of presets to ignore, including nested ones inside `extends`.',
+      'A list of presets to ignore, including any that are nested inside a `extends` array.',
```
Comment (rarkins) [inline lib/config/options/index.ts:167]:
```suggestion
      'A list of presets to ignore, including any that are nested inside an `extends` array.',
```

File: lib/config/options/index.ts [hunk 5]
```diff
@@ -509,7 +512,7 @@ const options: RenovateOptions[] = [
   {
     name: 'encrypted',
     description:
-      'A configuration object containing configuration encrypted with project key.',
+      'A configuration object containing configuration encrypted with project key.', // The word configuration is used twice here, not sure how to fix this line.
```
Comment (rarkins) [inline lib/config/options/index.ts:515]:
```suggestion
      'An object containing configuration encrypted with project key.',
```

File: lib/config/options/index.ts [hunk 6]
```diff
@@ -618,48 +620,49 @@ const options: RenovateOptions[] = [
   },
   {
     name: 'username',
-    description: 'Username for authentication. Currently Bitbucket only.',
+    description: 'Username for authentication. Currently Bitbucket only.', // Shouldn't we use the `supportedPlatforms` property for the Bitbucket message?
```
Comment (rarkins) [inline lib/config/options/index.ts:623]: Yes, but do we expose that field into our docs yet?

File: lib/config/options/index.ts [hunk 7]
```diff
@@ -618,48 +620,49 @@ const options: RenovateOptions[] = [
...
     stage: 'repository',
     type: 'string',
     globalOnly: true,
   },
   {
     name: 'password',
     description:
-      'Password for authentication. Currently Bitbucket only (AppPassword).',
+      'Password for authentication. Currently Bitbucket only (AppPassword).', // Shouldn't we use the `supportedPlatforms` property for the Bitbucket message?
```
Comment (rarkins) [inline lib/config/options/index.ts:631]: Same as above

File: lib/config/options/index.ts [hunk 8]
```diff
@@ -618,48 +620,49 @@ const options: RenovateOptions[] = [
...
     name: 'updateLockFiles',
-    description: 'Set to false to disable lock file updating.',
+    description: 'Set to `false` to disable lock file updating.',
     type: 'boolean',
   },
   {
     name: 'skipInstalls',
     description:
-      'Skip installing modules/dependencies if lock file updating is possible alone.',
+      'Skip installing modules/dependencies if lock file updating is possible alone.', // Possible alone is weird, need to find a better term
```
Comment (rarkins) [inline lib/config/options/index.ts:665]:
```suggestion
      'Skip installing modules/dependencies if lock file updating is possible without a full install.',
```

File: lib/config/options/index.ts [hunk 9]
```diff
@@ -1166,7 +1170,7 @@ const options: RenovateOptions[] = [
   {
     name: 'ignoreDeprecated',
     description:
-      'Ignore deprecated versions unless the current version is deprecated.',
+      'Ignore deprecated versions unless the current version is deprecated.', // I don't like this sentence. Needs a rewrite.
```
Comment (rarkins) [inline lib/config/options/index.ts:1173]:
```suggestion
      'Avoid upgrading from a non-deprecated version to a deprecated one.',
```

File: lib/config/options/index.ts [hunk 10]
```diff
@@ -1593,20 +1597,20 @@ const options: RenovateOptions[] = [
   {
     name: 'prTitle',
     description:
-      'Pull Request title template (deprecated). Now uses commitMessage.',
+      'Pull Request title template (deprecated). Now uses `commitMessage`.', // commitMessage is deprecated and not recommended, should that affect this message as well?
```
Comment (rarkins) [inline lib/config/options/index.ts:1600]:
```suggestion
      'Pull Request title template (deprecated). Inherits from `commitMessage` if null.',
```

File: lib/config/options/index.ts [hunk 11]
```diff
@@ -1961,7 +1965,7 @@ const options: RenovateOptions[] = [
   {
     name: 'abortIgnoreStatusCodes',
     description:
-      'A list of HTTP status codes to ignore and *not* abort the run because of when abortOnError=true.',
+      'A list of HTTP status codes to ignore and *not* abort the run because of when `abortOnError=true`.', // Sentence needs a rewrite also.
```
Comment (rarkins) [inline lib/config/options/index.ts:1968]:
```suggestion
      'A list of HTTP status codes safe to ignore even when `abortOnError=true`.',
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/config/options/index.ts (hunk 6 above)
Comment (HonkingGoose) [reply lib/config/options/index.ts:623]: It's exposed for some config options at least. Check out the docs for [`azureAutoApprove`](https://docs.renovatebot.com/configuration-options/#azureautoapprove):

azureAutoapprove-exposes-supportedPlatforms

---

## Do I need to change the docs generating code?

Do I need to change something in the docs generating code so it exposes `supportedPlatforms` for the `username` option?

Comment (rarkins) [reply lib/config/options/index.ts:623]: Seems like it's exposed so we can drop the extra description as you suggested

File: lib/config/options/index.ts (hunk 3 above)
Comment (rarkins) [reply lib/config/options/index.ts:129]: Potentially. I think we should either leave in `etc` or replace it with `comments`. There may also be links in status check comments

File: lib/config/options/index.ts [hunk 12]
```diff
@@ -101,31 +102,31 @@ const options: RenovateOptions[] = [
...
+      'Change this value to override the default onboarding PR title.',
     type: 'string',
     default: 'Configure Renovate',
     globalOnly: true,
     cli: false,
   },
   {
     name: 'productLinks',
-    description: 'Links which are embedded within PRs, issues, etc.',
+    description: 'Links which are used in PRs, issues and comments.',
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:129]: Check if this is what you wanted. 😉

File: lib/config/options/index.ts [hunk 13]
```diff
@@ -618,48 +620,50 @@ const options: RenovateOptions[] = [
   },
   {
     name: 'username',
-    description: 'Username for authentication. Currently Bitbucket only.',
+    description: 'Username for authentication.',
     stage: 'repository',
     type: 'string',
+    supportedPlatforms: ['bitbucket', 'bitbucket-server'],
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:626]: I've copy pasted the `supportedPlatforms` array from the `bbUseDefaultReviewers` config option in this file.

Please check if this is correct. 😉

File: lib/config/options/index.ts [hunk 14]
```diff
@@ -618,48 +620,50 @@ const options: RenovateOptions[] = [
...
     globalOnly: true,
   },
   {
     name: 'password',
-    description:
-      'Password for authentication. Currently Bitbucket only (AppPassword).',
+    description: 'Password for authentication.',
     stage: 'repository',
     type: 'string',
+    supportedPlatforms: ['bitbucket', 'bitbucket-server'],
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:634]: Same thing here: I've copy pasted the `supportedPlatforms` array from the `bbUseDefaultReviewers` config option in this file.

Please check if this is correct. 😉

File: lib/config/options/index.ts (hunk 13 above)
Comment (viceice) [reply lib/config/options/index.ts:626]:
```suggestion
    supportedPlatforms: ['azure', 'bitbucket', 'bitbucket-server'],
```

also supported on azure

File: lib/config/options/index.ts (hunk 14 above)
Comment (viceice) [reply lib/config/options/index.ts:634]:
```suggestion
    supportedPlatforms: ['azure', 'bitbucket', 'bitbucket-server'],
```

also supported on azure

**Review by viceice — COMMENTED** [review]
otherwise LGTM 🤷‍♂️

**Review by rarkins — APPROVED** [review]

**Pull request merged**